sdk/src/main/scala/io/horizen/history/AbstractHistory.scala

```diff
@@ -321,6 +320,28 @@ abstract class AbstractHistory[
     indexes :+ 1
   }
 
+  // calculate the other node approximate height knowing that the sequence was created using the method knownBlocksHeightToSync
+  protected def calculateOtherNodeApproxHeight(startingHeight: Int, sequenceSize: Int): Int = {
```

Comment:
Can you please add a test(unit will do) that will fail without the fix and pass with the fix

Reply:
Unit test added (`SidechainHistoryTest.synchronizationTestWithHighBlockNumber`) to simulate the task example. The number of blocks involved is less than the one of the task to keep the test runtime low but enough to manage divergent suffix sequence that involved height difference with exponential step.